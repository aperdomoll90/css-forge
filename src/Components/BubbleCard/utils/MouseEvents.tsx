export const rotateElement = (event: MouseEvent, element: Element | null) => {
  // get mouse position
  const x = event.clientX
  const y = event.clientY
  // find the middle
  const middleX = window.innerWidth / 2
  const middleY = window.innerHeight / 2

  //get offset from middle
  const offsetX = ((x - middleX) / middleX) * 25
  const offsetY = ((y - middleY) / middleY) * 25
  console.log('element', element)
  if (element) {
    element.style.setProperty('--rotateX', -1 * offsetY + 'deg')
    element.style.setProperty('--rotateY', offsetX + 'deg')
  }
}
