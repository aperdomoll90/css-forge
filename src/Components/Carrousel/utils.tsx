import * as React from 'react'


function getPrevElement(list:any) {
    const sibling = list[0].previousElementSibling
  
    if (sibling instanceof HTMLElement) {
      return sibling
    }
  
    return sibling
  }
  
  function getNextElement(list:any) {
    const sibling = list[list.length - 1].nextElementSibling
  
    if (sibling instanceof HTMLElement) {
      return sibling
    }
  
    return null
  }

  export default function usePosition(ref) {
    const [prevElement, setPrevElement] = React.useState(null)
    const [nextElement, setNextElement] = React.useState(null)
  
    React.useEffect(() => {
      const element = ref.current
  
      const update = () => {
        const rect = element.getBoundingClientRect()
  
        const visibleElements:any  = Array.from(element.children).filter(child => {
          const childRect = child.getBoundingClientRect()
  
          return childRect.left >= rect.left && childRect.right <= rect.right
        })
  
        if (visibleElements.length > 0) {
          setPrevElement(getPrevElement(visibleElements))
          setNextElement(getNextElement(visibleElements))
        }
      }
  
      update()
  
      element.addEventListener('scroll', update, { passive: true })
  
      return () => {
        element.removeEventListener('scroll', update, { passive: true })
      }
    }, [ref])
  
    const scrollToElement = React.useCallback(
      element => {
        const currentNode = ref.current
  
        if (!currentNode || !element) return
  
        let newScrollPosition
  
        newScrollPosition =
          element.offsetLeft +
          element.getBoundingClientRect().width / 2 -
          currentNode.getBoundingClientRect().width / 2
  
        currentNode.scroll({
          left: newScrollPosition,
          behavior: 'smooth',
        })
      },
      [ref]
    )
  
    const scrollRight = React.useCallback(
      () => scrollToElement(nextElement),
      [scrollToElement, nextElement]
    )
  
    const scrollLeft = React.useCallback(
      () => scrollToElement(prevElement),
      [scrollToElement, prevElement]
    )
  
    return {
      hasItemsOnLeft: prevElement !== null,
      hasItemsOnRight: nextElement !== null,
      scrollRight,
      scrollLeft,
    }
  }