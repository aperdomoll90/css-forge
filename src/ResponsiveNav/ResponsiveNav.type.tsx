import { menuItemsArrayPropsTypes } from '../utils/GlobalTypes.types'

export interface ResponsiveNavPropsTypes {
  menuItemsArray: menuItemsArrayPropsTypes[]
  height?: number
  width?: number
  logoHeight?: number
  logoMargin?: number
  primaryColor?: string
  secondaryColor?: string
  hoverColor?: string
  pressColor?: string
  labelColor?: string
}
