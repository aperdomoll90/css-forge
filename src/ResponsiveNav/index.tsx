export { ResponsiveNav } from "./ResponsiveNav"
export * from './ResponsiveNav.type'