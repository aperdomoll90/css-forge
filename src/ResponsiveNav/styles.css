.primary-header {
  height: var(--height);
  width: var(--width);
  align-items: center;
  position: absolute;
  inset: 0;
  justify-content: space-between;
}

.nav-flex-row {
  display: flex;
  gap: var(--gap, 1rem);
}

.nav-ff-sans-cond {
  font-family: 'Encode Sans Condensed', sans-serif;
}

.nav-uppercase {
  text-transform: uppercase;
}

.primary-logo {
  height: var(--logoHeight);
  width: auto;
  margin: var(--logoMargin);
}

.mobile-nav-toggle {
  display: none !important;
}

.primary-navigation {
  list-style: none;
  padding: 0;
  margin: 0;
  background: hsl(0 0% 0% / 0.75);
}
@supports (backdrop-filter: blur(1rem)) {
  .primary-navigation {
    background: hsl(0 0% 100% / 0.1);
    backdrop-filter: blur(1rem);
  }
}

.responsiveNav-link {
  text-decoration: none;
  color: var(--labelColor);
  font-weight: 300;
}

.responsiveNav-link > [aria-hidden='true'] {
  font-weight: 700;
  margin-inline-end: 0.5em;
}

@media (max-width: 35em) {
  .primary-navigation {
    --gap: 2em;
    position: fixed;
    z-index: 1000;
    inset: 0 0 0 30%;
    flex-direction: column;
    padding: min(30vh, 10rem) 2em;
    transform: translateX(100%);
    transition: transform 350ms ease-in;
  }
  .primary-navigation[data-visible='true'] {
    transform: translateX(0%);
  }
  .mobile-nav-toggle {
    display: flex !important;
  }
}

@media (min-width: 35em) and (max-width: 55em) {
  .primary-navigation a > span[aria-hidden] {
    display: none;
  }
}

@media (min-width: 35em) {
  .primary-navigation {
    --gap: clamp(1.5rem, 5vw, 3rem);
    padding-block: 1.5rem;
    padding-inline: clamp(3rem, 8vw, 10rem);
  }
}
