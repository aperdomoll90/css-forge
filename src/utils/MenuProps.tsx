import { menuItemsArrayPropsTypes } from './GlobalTypes.types'
import { BarIcon, CameraIcon, CodeIcon, GearIcon, HomeIcon, MapIcon, PawIcon, PlanetIcon, ProfileIcon } from '../IconCollection/IconCollection'
export const menuItemsArray: menuItemsArrayPropsTypes[] = [
  {
    label: 'Home',
    icon: HomeIcon,
    link: '#',
  },
  {
    label: 'Settings',
    icon: GearIcon,
    link: '#',
  },
  {
    label: 'Data',
    icon: BarIcon,
    link: '#',
  },
  {
    label: 'Profile',
    icon: ProfileIcon,
    link: '#',
  },
  {
    label: 'Code',
    icon: CodeIcon,
    link: '#',
  },
]

export const extendedMenuItemsArray: menuItemsArrayPropsTypes[] = [
    ...menuItemsArray,
    {
      label: 'Map',
      icon: MapIcon,
      link: '#',
    },
    {
      label: 'Animations',
      icon: CameraIcon,
      link: '#',
    },
    {
      label: 'Pets',
      icon: PawIcon,
      link: '#',
    },
  ]