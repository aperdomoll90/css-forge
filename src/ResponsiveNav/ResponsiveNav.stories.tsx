import { Meta, StoryFn } from '@storybook/react'
import { ResponsiveNav } from './ResponsiveNav'
import StorybookContainer from '../StorybookContainer'
import { ResponsiveNavPropsTypes } from './ResponsiveNav.type'
import { menuItemsArray } from '../utils/MenuProps'
import { colorPickerControl, pentaDecimals } from '../utils/StoryProps'
import logo from '../media/Logo.png'

export default {
  title: 'Navigation Bars',
  component: ResponsiveNav,
  parameters: {
    jest: ['ResponsiveNav.test.tsx'],
  },
  argTypes: {
    height: { options: pentaDecimals, control: { type: 'select' } },
    width: { options: pentaDecimals, control: { type: 'select' } },
    primaryColor: colorPickerControl,
    secondaryColor: colorPickerControl,
    labelColor: colorPickerControl,
    hoverColor: colorPickerControl,
    pressColor: colorPickerControl,
  },
} as Meta

const ResponsiveNavTemplate: StoryFn<ResponsiveNavPropsTypes> = (args) => (
  <StorybookContainer backgroundImage>
    <ResponsiveNav {...args} />
  </StorybookContainer>
)

export const ResponsiveNavComponent = ResponsiveNavTemplate.bind({})
ResponsiveNavComponent.args = {
  logo:logo,
  menuItemsArray: menuItemsArray,
}