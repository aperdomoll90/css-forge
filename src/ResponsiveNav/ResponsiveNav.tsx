import { useState } from 'react'
import './styles.css'
import '../utils/GlobalStyles.css'
import { ResponsiveNavPropsTypes } from './ResponsiveNav.type'
import { menuItemsArrayPropsTypes } from '../utils/GlobalTypes.types'
import { ToggleButton } from '../ToggleButton'

export const ResponsiveNav: React.FC<ResponsiveNavPropsTypes> = ({ logo, height, width, logoHeight, logoMargin, menuItemsArray, primaryColor, secondaryColor, hoverColor, pressColor, labelColor }) => {
  const [visible, setVisible] = useState(false)

  const stylesProps = {
    '--height': height ? `${height}rem` : '5.4rem',
    '--width': width ? `${width}rem` : '100%',
    '--logoHeight': logoHeight ? logoHeight : '3rem',
    '--logoMargin': logoMargin ? logoMargin : '1rem',
    '--primaryColor': primaryColor ? primaryColor : '#222327',
    '--secondaryColor': secondaryColor ? secondaryColor : '#29fd53',
    '--labelColor': labelColor ? labelColor : '#fff',
    '--hoverColor': hoverColor ? hoverColor : '#1c2942',
    '--pressColor': pressColor ? pressColor : '#1c2942',
  }

  const renderMenuItems = menuItemsArray?.map((item: menuItemsArrayPropsTypes, index: number) => (
    <li key={`${item.label}-${index}`} className='responsiveNav-active'>
      <a className='responsiveNav-link' href={item.link}>
        <span aria-hidden='true'>0{index}</span>
        {item.label}
      </a>
    </li>
  ))

  return (
    <section id='primary-header' className='primary-header flex-row' style={stylesProps as React.CSSProperties}>
      <div>{logo && <img src={logo} className='primary-logo' />}</div>
      <ToggleButton
        top={1}
        right={1}
        customClass='mobile-nav-toggle'
        ariaControls='primary-navigation'
        ariaExpanded={visible}
        active={visible}
        setActive={() => {
          setVisible(!visible)
        }}
        buttonBackgroundColor='transparent'
      />
      <nav>
        <ul id='primary-navigation' aria-aria-expanded={visible} data-visible={visible} className='primary-navigation uppercase ff-sans-cond flex-row'>
          {renderMenuItems}
        </ul>
      </nav>
    </section>
  )
}
