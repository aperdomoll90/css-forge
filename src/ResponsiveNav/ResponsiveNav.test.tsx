import { render } from '@testing-library/react'
import { ResponsiveNav } from './ResponsiveNav'
import { menuItemsArray } from '../utils/MenuProps'

describe('ResponsiveNav component', () => {
  const props = {
    menuItemsArray: menuItemsArray,
    height: 3,
    width: 5,
    primaryColor: 'rgb(0, 0, 0)',
    secondaryColor: 'rgb(0, 0, 0)',
    hoverColor: 'rgb(0, 0, 0)',
    pressColor: 'rgb(0, 0, 0)',
    labelColor: 'rgb(0, 0, 0)',
  }

  const renderComponent = () => render(<ResponsiveNav {...props} />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
