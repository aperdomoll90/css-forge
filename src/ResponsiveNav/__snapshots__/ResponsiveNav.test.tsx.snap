// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveNav component should match snapshots 1`] = `
<div>
  <section
    class="primary-header flex-row"
    id="primary-header"
    style="--height: 3rem; --width: 5rem; --logoHeight: 3rem; --logoMargin: 1rem; --primaryColor: rgb(0, 0, 0); --secondaryColor: rgb(0, 0, 0); --labelColor: rgb(0, 0, 0); --hoverColor: rgb(0, 0, 0); --pressColor: rgb(0, 0, 0);"
  >
    <div />
    <button
      aria-controls="primary-navigation"
      aria-expanded="false"
      class="ToggleButton mobile-nav-toggle"
      style="--size: 2rem; --color: #fff; --buttonHover: 0; --buttonBackgroundColor: transparent; --top: 1rem; --bottom: auto; --left: auto; --right: 1rem;"
    >
      <span />
      <span />
      <span />
      <span
        class="sr-only"
      >
        menu
      </span>
    </button>
    <nav>
      <ul
        aria-aria-expanded="false"
        class="primary-navigation uppercase ff-sans-cond flex-row"
        data-visible="false"
        id="primary-navigation"
      >
        <li
          class="responsiveNav-active"
        >
          <a
            class="responsiveNav-link"
            href="#"
          >
            <span
              aria-hidden="true"
            >
              0
              0
            </span>
            Home
          </a>
        </li>
        <li
          class="responsiveNav-active"
        >
          <a
            class="responsiveNav-link"
            href="#"
          >
            <span
              aria-hidden="true"
            >
              0
              1
            </span>
            Settings
          </a>
        </li>
        <li
          class="responsiveNav-active"
        >
          <a
            class="responsiveNav-link"
            href="#"
          >
            <span
              aria-hidden="true"
            >
              0
              2
            </span>
            Data
          </a>
        </li>
        <li
          class="responsiveNav-active"
        >
          <a
            class="responsiveNav-link"
            href="#"
          >
            <span
              aria-hidden="true"
            >
              0
              3
            </span>
            Profile
          </a>
        </li>
        <li
          class="responsiveNav-active"
        >
          <a
            class="responsiveNav-link"
            href="#"
          >
            <span
              aria-hidden="true"
            >
              0
              4
            </span>
            Code
          </a>
        </li>
      </ul>
    </nav>
  </section>
</div>
`;
